src/main.rs

```diff
@@ -323,21 +329,25 @@ impl SpiDrv {
     }
 
     // TODO: replace last_param with an enumerated type, e.g. NO_LAST_PARAM, LAST_PARAM
-    fn send_param(&mut self, uart: &mut EnabledUart, param: u8, param_len: u8, last_param: bool) -> SpiResult<()> {
+    fn send_param(&mut self, uart: &mut EnabledUart, params: &mut Params, last_param: bool) -> SpiResult<()> {
+        let param_len: u8 = params.len() as u8;
         let res = self.send_param_len8(uart, param_len);
         match res {
             Ok(_) => {
                 // TODO: this doesn't quite match the C++ code yet, seems it can send a
                 // variable length buf
-                let byte_buf = &mut[param];
-                write!(uart, "\t\tsending byte: 0x{:X?} -> ", param).ok().unwrap();
+                // let mut byte_buf: [u8; param_len] =  params.into_array().unwrap();
+                // let byte_buf = &mut[params[0]];
+                let byte_buf = params;
+                //write!(uart, "\t\tsending bytes: 0x{:X?} -> ", params).ok().unwrap();
```

Comment:
Maybe remove some of these commented out experiments? I feel like on the one hand it's nice to have some history on things that we've tried. On the other hand It has the potential to lead us to continue down dead end paths by influencing our thought process... I'm very open to what ya'll think about this.

```suggestion
                let byte_buf = params;
```

Reply:
Sounds good. Removed that part